### PR TITLE
[Assignment Tracking] Change placeholder from `undef` to `poison`

### PR DESCRIPTION
**Pull request by pedroclobo**
Empty metadata address components of `dbg.assign` can be represented as `poison` values instead of `undef` values.

### CODE REVIEW

**Comment by pedroclobo** [comment]
cc @nunoplopes

File: llvm/lib/IR/IntrinsicInst.cpp [hunk 1]
```diff
@@ -223,12 +223,12 @@ void DbgAssignIntrinsic::setAddress(Value *V) {
...
   if (isKillAddress())
     return;
-  setAddress(UndefValue::get(getAddress()->getType()));
+  setAddress(PoisonValue::get(getAddress()->getType()));
 }
 
 bool DbgAssignIntrinsic::isKillAddress() const {
   Value *Addr = getAddress();
-  return !Addr || isa<UndefValue>(Addr);
+  return !Addr || isa<PoisonValue>(Addr);
```
Comment (nunoplopes) [inline llvm/lib/IR/IntrinsicInst.cpp:231]: don't change this one yet for compatibility with older IR.

Comment (pedroclobo) [reply llvm/lib/IR/IntrinsicInst.cpp]: Fixed!

**Comment by nunoplopes** [comment]
seems like a spurious failure; the build bot is green again.

**Pull request merged**